packages/dev/src/main.ts

```diff
@@ -486,8 +486,15 @@ export class NetlifyDev {
         ),
       }
 
+      // Use mock mode in test environment to ensure deterministic tests
+      const isTest = typeof process !== 'undefined' && (
+        process.env.NODE_ENV === 'test' ||
+        process.env.VITEST === 'true' ||
+        process.env.npm_lifecycle_event === 'test'
+      )
```

Comment:
wtf, no. remove this. instead, add a `geolocation.mode` option to the edge function feature, that defaulst to `cache`. then in dev pkg tests set it to `mock`.